ebean-core/src/main/java/io/ebeaninternal/server/core/HtmlMetricReportGenerator.java

```diff
@@ -0,0 +1,679 @@
+    tab.startTable("Name", "Value");
+    tab.input("initRequest.thresholdMicros", initRequest.thresholdMicros());
+    tab.input("initRequest.isAll", initRequest.isAll() ? 1 : 0);
+    tab.action("initRequest.apply", "Start capturing");
+
+    tab.input("queryRequest.maxCount", queryRequest.maxCount());
+    tab.input("queryRequest.maxTimeMillis", queryRequest.maxTimeMillis());
+    tab.input("queryRequest.since", queryRequest.since());
+    tab.tableRow("current time", System.currentTimeMillis());
+    tab.action("queryRequest.apply", "Collect plans");
+
+
+    tab.action("clearCaches", "Clear caches");
+    tab.action("resetMetrics", "Reset all metrics");
+
+    tab.endTable();
+    tab.html("<h2>Usage:</h2>");
+    tab.html("<ol>\n");
+    tab.html("<li>Set initRequest.thresholdMicros to the value, you want to capture</li>\n");
+    tab.html("<li>Add hashes or set initRequest.isAll = 1 (default)</li>\n");
+    tab.html("<li>Click 'Start capturing'</li>\n");
+    tab.html("<li>Do the action, which executes the query/queries you want to inspect</li>\n");
+    tab.html("<li>Click 'Collect plans' to collect the query-plans</li>\n");
+    tab.html("</ol>\n");
+  }
+
+  /**
+   * 2nd tab: Query Metrics.
+   */
+  protected void queryMetricTab(List<MetaQueryMetric> metrics, Html html) {
+
+    HtmlTab tab = html.tab("Query metrics");
+    tab.startTable("type?", "?", "?", "?", "count", "total", "mean", "max", "sql", "hash");
+    for (MetaQueryMetric metric : metrics) {
+      String[] names = splitPad(metric.name(), 4);
+      tab.tableRow(names[0], names[1], names[2], names[3],
+        metric.count(),
+        micros(metric.total()), micros(metric.mean()), micros(metric.max()),
+        metric.sql(), hash(metric.hash()));
+    }
+    tab.endTable();
+  }
+
+  /**
+   * 3rd tab: Timed metrics.
+   */
+  protected void timedMetricTab(List<MetaTimedMetric> metrics, Html html) {
+    HtmlTab tab = html.tab("Timed metrics");
+    tab.startTable("type?", "?", "?", "?", "count", "total", "mean", "max");
+    for (MetaTimedMetric metric : metrics) {
+      String[] names = splitPad(metric.name(), 4);
+      tab.tableRow(names[0], names[1], names[2], names[3],
+        metric.count(),
+        micros(metric.total()), micros(metric.mean()), micros(metric.max()));
+    }
+    tab.endTable();
+  }
+
+  /**
+   * 4th tab: Count metrics.
+   */
+  protected void countMetricTab(List<MetaCountMetric> metrics, Html html) {
+    HtmlTab tab = html.tab("Count Metrics");
+    tab.startTable("type?", "?", "?", "?", "count");
+    for (MetaCountMetric metric : metrics) {
+      String[] names = splitPad(metric.name(), 4);
+      tab.tableRow(names[0], names[1], names[2], names[3], metric.count());
+    }
+    tab.endTable();
+  }
+
+  /**
+   * 5th tab: Query plans.
+   */
+  protected void queryPlansTab(Html html) {
+    HtmlTab tab = html.tab("Query plans");
+    tab.startTable("type?", /*"tenant?",*/ "count", "micros", "sql", "hash");
```

Comment:
Important. This depends on #2955 and should be changed, either which one is merged first